app/controllers/concerns/wallet_func.rb

```diff
@@ -0,0 +1,18 @@
+module WalletFunc
+  def create_random_wallet_number
+    rand(1_111_111_111_111_111..9_999_999_999_999_999)
+  end
+
+  def currencies
+    Currency.all
+  end
+
+  def unlocked_user_wallets(user)
+    wallets = []
+
+    user.wallets.each do |wallet|
+      wallets << wallet if wallet.locked == false
+    end
+    wallets
+  end
```

Comment:
```suggestion
  def unlocked_user_wallets(user_id)
    Wallet.where(user_id: user_id, locked: false)
  end
```
it will be much better and clearer this way